### DISCUSSION TITLE
Max recording size and stream time limit

### DISCUSSION THREAD

**Discussion by ankit-akteena** [ant-media/Ant-Media-Server#6858] [Q&A] [answered]
Hello,

I have a couple of questions:

1.Is there a limit on the duration of a live stream? For example, does the system impose any restrictions on continuous streaming, or can it support long-duration streams, such as 8–10 hours, without interruption? Additionally, are there any recommended practices for handling extended live streams to ensure stability and performance?

2.What is the maximum recording size supported? I couldn’t find any limit mentioned in the documentation. Additionally, in terms of latency, if I have a 1-hour stream recording, how long would it take to upload to my S3 bucket from Ant Media?

**Comment by Mohit-3196** [comment, marked as answer]
Hi @ankit-akteena,
I hope you are doing well.

1. There is no limitation for the duration of the live stream. It can be of any duration & there are no server-side restrictions. It's about the resources & they depend upon the use case. It is recommended to have atleast a 4 cores CPU optimized instance with 8GB RAM for running the Ant Media Server enterprise edition.
2. There is no limit on the recording size as well. If you are storing it on the server, then you should have enough storage on the server. For the S3 uploads, if you are recording the stream with MP4 then they are uploaded to the S3 after the stream is ended or the recording is stopped, post the MP4 mixing process.
For HLS recording (recommended for long duration), they can be uploaded to the S3 in real time.

I hope I was able to answer your questions.

We can also meet & discuss about your use case & show you that everything is working fine as you need :)

--
Cheers,
Mohit Dubey

**Reply by ankit-akteena** [reply]
Thanks for the reply @Mohit-3196 , can you please take a look at this doubt as well https://github.com/orgs/ant-media/discussions/6853

**Reply by Mohit-3196** [reply]
Sure @ankit-akteena :)
